src/cargo/ops/lockfile.rs

```diff
@@ -19,11 +19,15 @@ pub fn load_pkg_lockfile(pkg: &Package, config: &Config)
 
 pub fn load_lockfile(path: &Path, pkg: &Package, config: &Config)
                      -> CargoResult<Option<Resolve>> {
-    // If there is no lockfile, return none.
-    let mut f = match File::open(path) {
-        Ok(f) => f,
-        Err(_) => return Ok(None)
-    };
+
+    let lockfile = pkg.root().join("Cargo.lock");
+
+    // If there is no lockfile, generate one.
+    if !lockfile.exists() {
+       let _ = generate_lockfile(pkg.manifest_path(), config);
```

Comment:
This affects other commands so this logic should probably stay inside of `update_lockfile`